synapse/datamodel.py

```diff
@@ -295,10 +292,16 @@ def getPropNorm(self, prop, valu, oldval=None):
 
         '''
         dtype = self.getPropType(prop)
-        if dtype == None:
-            return valu
+        if dtype != None:
+            try:
+                valu = dtype.norm(valu,oldval=oldval)
+            except BadTypeValu:
+                raise BadPropValu(name=prop, valu=valu)
+
+        if not s_compat.issystype(valu):
```

Comment:
At this point, it's probably best to fall back to the enforcement that happens at the rows level API ( which would be required to protect the storage layers anyway ) than to have all the types implementations each manage that themselves.  For now, i think all changes to datamodel.py should be reverted ( since we haven't finalized the discussion on coersion of bool to int just yet )